src/main/java/net/rubygrapefruit/platform/internal/jni/AbstractFileEventFunctions.java

```diff
@@ -96,16 +99,26 @@ public void run() {
 
         @Override
         public void startWatching(Collection<File> paths) {
-            ensureOpen();
-            startWatching0(server, toAbsolutePaths(paths));
+            closeLock.readLock().lock();
```

Comment:
I am not sure we need to make the `FileWatcher` here thread safe. I think it is perfectly fine if it isn't thread safe.

Reply:
I reverted that change.